src/glum/_glm.py

```diff
@@ -1873,8 +1863,70 @@ def _expand_categorical_penalties(penalty, X, drop_first):
         elif isinstance(X, np.ndarray):
             X = tm.DenseMatrix(X)
 
+        if not hasattr(self, "feature_names_") or not hasattr(self, "term_names_"):
+            self._get_feature_names(X)
```

Comment:
Given that feature names are always created, why do they need to be set separately for pd.DataFrames and other objects? I.e., why not move `self._get_feature_names(X)` in L. 1749 out of the condition to line 1746, say?

Reply:
That was my first attempt, but the problem is that at that point it is not certain that `X` is a nice, 2-dimensional array-like object. In particular, `sklearn.utils.estimator_check.check_estimator` is part of the test suite, and it feeds our estimator all kinds of weird objects that `sklearn.utils.check_array` is supposed to handle.